### PR TITLE
feat: add chaos mesh infra

### PR DESCRIPTION
**Pull request by egermaneau-aneo**
**added Chaos Mesh**

I created :

- infrastructure/modules/k8s/chaos-mesh
- infrastructure/quick-deploy/{localhost,gcp}/all-in-one/chaos-mesh.tf

Updated : 

- infrastructure/quick-deploy/{localhost,gcp}/all-in-one/{variables,outputs}
- versions.tfvars.json

_Would be good to do a squash merge_

### CODE REVIEW

File: infrastructure/modules/k8s/chaos-mesh/main.tf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  count = var.create_namespace ? 1 : 0
+  metadata {
+    name = var.namespace
+  }
+}
+
+resource "helm_release" "chaosmesh" {
+  depends_on = [kubernetes_namespace.chaosmesh]
+  name       = "chaosmesh"
+  namespace  = var.namespace
```
Comment (lemaitre-aneo) [inline infrastructure/modules/k8s/chaos-mesh/main.tf:12]:
```suggestion
  name       = "chaosmesh"
  namespace  = kubernetes_namespace.chaosmesh.metadata[0].namespace
```

File: infrastructure/modules/k8s/chaos-mesh/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+output "chaos-mesh" {
```
Comment (lemaitre-aneo) [inline infrastructure/modules/k8s/chaos-mesh/outputs.tf:1]: You are alrady in the module chaos_mesh, so I would just call the output chart_name and inline the value.

File: infrastructure/quick-deploy/gcp/all-in-one/variables.tf [hunk 3]
```diff
@@ -90,6 +90,25 @@ variable "keda" {
   default = {}
 }
 
+# Chaos Mesh
+variable "chaos-mesh" {
+  description = "Chaos Mesh configuration"
+  type = object({
+    namespace                 = optional(string, "chaos-mesh")
+    chaosmesh_image_name      = optional(string, "ghcr.io/chaos-mesh/chaos-mesh"),
+    chaosmesh_image_tag       = optional(string, "v2.6.3"),
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/gcp/all-in-one/variables.tf:99]: Please use the `versions.tfvars.json` to provide the default tag.
For GCP, you would need to add the image to the GAR: https://github.com/aneoconsulting/ArmoniK/blob/main/infrastructure/quick-deploy/gcp/all-in-one/gar.tf

File: infrastructure/quick-deploy/gcp/all-in-one/chaos-mesh.tf [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+data "external" "get_chaos_dashboard_port" {
+  program    = ["sh", "-c", "kubectl get svc chaos-dashboard -n chaos-mesh -o jsonpath='{\"{\"}\"nodeport\": \"{.spec.ports[0].nodePort}\"}'"]
+  depends_on = [module.chaos-mesh]
+}
+
+data "external" "get_chaos_dashboard_ip" {
+  program    = ["sh", "-c", "kubectl get svc chaos-dashboard -n chaos-mesh -o jsonpath='{\"{\"}\"ip\": \"{.status.loadBalancer.ingress[0].ip}\"}'"]
+  depends_on = [module.chaos-mesh]
+}
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/gcp/all-in-one/chaos-mesh.tf:33]: This should most likely be in the chaos-mesh module.

File: infrastructure/modules/k8s/chaos-mesh/outputs.tf (hunk 2 above)
Comment (egermaneau-aneo) [reply infrastructure/modules/k8s/chaos-mesh/outputs.tf]: done

File: infrastructure/quick-deploy/gcp/all-in-one/chaos-mesh.tf (hunk 4 above)
Comment (egermaneau-aneo) [reply infrastructure/quick-deploy/gcp/all-in-one/chaos-mesh.tf]: done

**Review by egermaneau-aneo — COMMENTED** [review]
Think it's all done

**Review by egermaneau-aneo — COMMENTED** [review]
Seems ok now.
The on;y issue is about node selector affectation in set blocks.

**Comment by lemaitre-aneo** [comment]
> Seems ok now. The on;y issue is about node selector affectation in set blocks.

I think we could do it with the values property instead.

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**